src/types/BulkUpload.ts

```diff
@@ -0,0 +1,41 @@
+import { ajv } from '../ajv';
+import type { JSONSchemaType } from 'ajv';
+
+export enum BulkUploadStatus {
+  PENDING = 'pending',
+  IN_PROGRESS = 'in_progress',
+  COMPLETED = 'completed',
+  FAILED = 'failed',
+  CANCELED = 'canceled',
+}
+
+export interface BulkUpload {
+  readonly id: number;
+  fileName: string
+  sourceUrl: string;
+  status: BulkUploadStatus;
+  readonly createdAt: Date;
+}
+
+// See https://github.com/typescript-eslint/typescript-eslint/issues/1824
+/* eslint-disable @typescript-eslint/indent */
+export type BulkUploadCreate = Omit<BulkUpload, 'createdAt' | 'status' | 'id'>;
+/* eslint-enable @typescript-eslint/indent */
+
+export const bulkUploadCreateSchema: JSONSchemaType<BulkUploadCreate> = {
+  type: 'object',
+  properties: {
+    fileName: {
+      type: 'string',
+    },
+    sourceUrl: {
+      type: 'string',
+    },
```

Comment:
Out of curiosity, is this where you would put validation around the source? Or would you do that in code rather than JSONSchema? (Feel free to say "I'll figure that out when we get there", of course!)

Reply:
Good question!

I'd probably want to use the json schema for it if possible (in the spirit of "AJV is our validation engine and it is driven by the json schema").  That way an invalid URL is rejected on POST.

I probably *also* want to defensively validate the URL before invoking any kind of fetch...

I do think it'd be appropriate to add that validation on that first part as part of this PR.

Reply:
Cool - in that case, I have a pending PR around S3 buckets that might be relevant here; we can decide how we want to sequence it.

Reply:
Ok so I think there's two categories of validation here and I'm a little torn.

One is on the general form -- e.g. "this should be a URL" -- which I think belongs in the schema.

The other might be further constraint that is more complex / dynamic -- e.g. "this should be a URL on { DOMAIN X }" -- that I'm a little bit wary of baking into this schema definition because we might have a world where these schemas are driven by the (non-dynamic) swagger spec...

For now I'm going to add validation about the string structure (that it's URL), and will punt in the short term on the trickier validation question.

Reply:
Perfect!